state-chain/chains/src/lib.rs

```diff
@@ -977,3 +982,116 @@ pub trait DepositDetailsToTransactionInId<C: ChainCrypto> {
 		None
 	}
 }
+
+#[derive(
+	Clone, Debug, Encode, Decode, PartialEq, Eq, TypeInfo, Serialize, Deserialize, PartialOrd, Ord,
+)]
+pub enum GenericVaultSwapExtraParameters<Address, Number> {
+	Btc {
+		min_output_amount: Number,
+		retry_duration: BlockNumber,
+	},
+	Sol {
+		from: Address,
+		event_data_account: Address,
+		input_amount: Number,
+		refund_parameters: ChannelRefundParametersGeneric<Address>,
+	},
+	SolUsdc {
+		from: Address,
+		from_token_account: Address,
+		event_data_account: Address,
+		input_amount: Number,
```

Comment:
Yeah its because for RPC we use NumberOrHex type, but internally we use AssetAmount.
Having the Number as generic allow the same type to be used everywhere